pandas/plotting/_matplotlib/boxplot.py

```diff
@@ -150,7 +154,7 @@ def _make_plot(self):
                 labels = [pprint_thing(key) for key in range(len(labels))]
             self._set_ticklabels(ax, labels)
 
-    def _set_ticklabels(self, ax, labels):
+    def _set_ticklabels(self, ax: "Axes", labels):
```

Comment:
I've opened #36034 so that when reviewing we can  just click on and get definition as doesn't work for string literals (in VS Code anyway)